sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java

```diff
@@ -140,6 +149,66 @@ public void testNullSubscription() {
     assertNotNull(DisplayData.from(read));
   }
 
+  /**
+   * {@link PipelineOptions} with a ValueProvider.
+   */
+  public interface RuntimeSubscriptionOptions extends PipelineOptions {
+    ValueProvider<String> getValue();
+    void setValue(ValueProvider<String> provider);
+  }
+
+  @Test
+  public void testValueProviderSubscription() {
+    RuntimeSubscriptionOptions options =
+        PipelineOptionsFactory.create().as(RuntimeSubscriptionOptions.class);
+    options.setValue(StaticValueProvider.of("projects/project/subscriptions/subscription"));
+    ValueProvider<String> subscription = options.getValue();
+    Read<String> pubsubRead = PubsubIO.readStrings().fromSubscription(subscription);
+    Pipeline.create().apply(pubsubRead);
+    assertThat(pubsubRead.getSubscriptionProvider(), not(nullValue()));
+    assertThat(pubsubRead.getSubscriptionProvider().isAccessible(), is(true));
+    assertThat(
+        pubsubRead.getSubscriptionProvider().get().asPath(),
+        equalTo(options.getValue().get()));
+  }
+
+  @Test
+  public void testRuntimeValueProviderSubscription() {
+    RuntimeSubscriptionOptions options =
+        PipelineOptionsFactory.create().as(RuntimeSubscriptionOptions.class);
+    ValueProvider<String> subscription = options.getValue();
+    Read<String> pubsubRead = PubsubIO.readStrings().fromSubscription(subscription);
+    Pipeline.create().apply(pubsubRead);
+    assertThat(pubsubRead.getSubscriptionProvider(), not(nullValue()));
+    assertThat(pubsubRead.getSubscriptionProvider().isAccessible(), is(false));
+  }
+
+  @Test
+  public void testValueProviderTopic() {
+    RuntimeSubscriptionOptions options =
+        PipelineOptionsFactory.create().as(RuntimeSubscriptionOptions.class);
+    options.setValue(StaticValueProvider.of("projects/project/topics/topic"));
+    ValueProvider<String> topic = options.getValue();
+    Read<String> pubsubRead = PubsubIO.readStrings().fromTopic(topic);
+    Pipeline.create().apply(pubsubRead);
+    assertThat(pubsubRead.getTopicProvider(), not(nullValue()));
+    assertThat(pubsubRead.getTopicProvider().isAccessible(), is(true));
+    assertThat(
+        pubsubRead.getTopicProvider().get().asPath(),
+        equalTo(options.getValue().get()));
+  }
+
+  @Test
+  public void testRuntimeValueProviderTopic() {
+    RuntimeSubscriptionOptions options =
+        PipelineOptionsFactory.create().as(RuntimeSubscriptionOptions.class);
+    ValueProvider<String> topic = options.getValue();
+    Read<String> pubsubRead = PubsubIO.readStrings().fromTopic(topic);
+    Pipeline.create().apply(pubsubRead);
+    assertThat(pubsubRead.getTopicProvider(), not(nullValue()));
+    assertThat(pubsubRead.getTopicProvider().isAccessible(), is(false));
+  }
+
```

Comment:
Seems like we still don't have runnable pipeline tests for PubsubIO?..

Reply:
That is my understanding yes.